src/lib/classes/notification.js

```diff
@@ -0,0 +1,512 @@
+import solid from 'solid-auth-client';
+import N3 from 'n3';
+import solidLDflex from '@solid/query-ldflex';
+import { solidResponse, SolidError, getBasicPod } from '@utils';
+import defaultShape from '../shapes/notification.json';
+
+const PREFIXES = {
+  terms: 'https://www.w3.org/ns/solid/terms#',
+  schema: 'http://www.w3.org/2000/01/rdf-schema#',
+  things: 'https://schema.org/Thing#',
+  ns: 'https://www.w3.org/1999/02/22-rdf-syntax-ns#',
+  foaf: 'http://xmlns.com/foaf/0.1/',
+  acl: 'http://www.w3.org/ns/auth/acl#',
+  ldp: 'http://www.w3.org/ns/ldp#'
+};
+
+/**
+ * Notification Class for SOLID
+ * To know more about ACL please go to: https://github.com/solid/web-access-control-spec
+ * To know more about N3 please go to: https://www.npmjs.com/package/n3
+ */
+export class Notification {
+  constructor(owner) {
+    if (Notification.instance) {
+      return Notification.instance;
+    }
+    this.shape = defaultShape;
+    this.owner = owner;
+    this.schema = null;
+    this.shapeName = 'default';
+    Notification.instance = this;
+  }
+
+  setOwner = owner => {
+    this.owner = owner;
+  };
+
+  /**
+   * We are checking if the user has an inbox reference on the card and, also, if the inbox folder exists.
+   * @param path
+   * @returns {Promise<*|boolean>}
```

Comment:
I think that should be Promise<>?

Reply:
should be Promise<boolean> I will fix it